aten/src/ATen/test/cpu_rng_test.cpp

```diff
@@ -139,9 +127,7 @@ TORCH_LIBRARY_IMPL(aten, CustomRNGKeyId, m) {
   m.impl("normal.Tensor_float_out",  normal_Tensor_float_out);
   m.impl("normal.float_Tensor_out",  normal_float_Tensor_out);
   m.impl("normal.Tensor_Tensor_out", normal_Tensor_Tensor_out);
-  m.impl("normal.Tensor_float",      normal_Tensor_float);
-  m.impl("normal.float_Tensor",      normal_float_Tensor);
-  m.impl("normal.Tensor_Tensor",     normal_Tensor_Tensor);
```

Comment:
How come just half of them were deleted? In fact, is this necessary given that now these operations are implemented as structured kernels? cc @ysiraichi  @peterbell10

Reply:
it's because these previously used the template version directly (to pass a custom rng for testing and demo this functionality). not sure what to do here since these are gone now

Reply:
I think the distribution templates are part of the public generator API, so can't be removed. e.g. the cryptographic PRNG uses them:
https://github.com/pytorch/csprng/blob/5a6d9458c142190d5d713744687434c73c06ad01/torchcsprng/csrc/kernels_body.inc#L257

Reply:
What do you think we should do here @mruberry ? Should we do the checks twice in these functions?
See https://github.com/pytorch/pytorch/pull/69628#discussion_r765525408 for context.

Reply:
I think, ideally, we'd want those checks in the META function. One way out of this is to factor out the implementation (code after the type checks) into a new function. Then, we would have something like this:

- `normal_impl_out`: dtype checks and calls `normal_impl_impl_out` (not a very good name)
- `normal_impl_impl_out`: executes the rest of the implementation

Then, the IMPL function can just call `normal_impl_impl_out` directly, which would bypass dtype checks (these can be factored into a function of its own, and called in META, too).

Not sure whether the extra indirection is worth it, though.

Reply:
@bdhirsh will take a look soon -- I think he's the best person to help answer this question

Reply:
cc @pbelevich (who I think wrote the rng api).

It looks like the templates are public API to help write out-of-tree kernel extensions for distribution ops, so we can't get easily rid of them (without finding all external usages and making them structured too, which... would require external codegen and doesn't seem super beneficial to do). There's also more context described [here](https://github.com/pytorch/pytorch/blob/3d358a76783acef87f282cc0fb20818eeb4c133c/c10/core/DispatchKey.h#L101). 

If that's right, then I don't think that porting `normal_*` ops to structured will really help to clean up much code - we have to keep all of the functional/inplace/out= template variants around. @ysiraichi is also right, you'd need to make sure that all of the error checking logic currently in the template is run in the meta function (and also directly in the template, since out-of-tree kernel writers still need to rely on them).

Given all of that, it sounds to me like it would be easiest to just directly write meta kernels for all of the distribution ops.

@ysiraichi also pointed out that `normal()` has some special resize handling

Reply:
@bdhirsh 

> Given all of that, it sounds to me like it would be easiest to just directly write meta kernels for all of the distribution ops.

I'm going to do this and will create a new stack with the changes. This stack will stay open for now for reference.

I'll also fix the broadcasting issue that I introduced, which breaks BC.